py/picca/pk1d/postproc_pk1d.py

```diff
@@ -0,0 +1,331 @@
+                    for stats in stats_array:
+                        meanP1D_table[stats+c][index] = np.nan
+                    continue
+
+                if weight_method=='fit_snr':
+                    snr_bin_edges = np.arange(1,11,1)  # 9 SNR bins are used for the fit
+                    snr_bins = (snr_bin_edges[:-1]+snr_bin_edges[1:])/2
+                    def variance_function(snr, a, b):
+                        return (a/(snr-1)**2) + b
+                    data_values = p1d_table[c][select]
+                    data_snr = p1d_table['forest_snr'][select]
+                    # Fit function to observed dispersion:
+                    standard_dev,_,_ = binned_statistic(data_snr, data_values,
+                                                                 statistic='std', bins=snr_bin_edges)
+                    coef, coef_cov = curve_fit(variance_function, snr_bins, standard_dev**2, bounds=(0,np.inf))
+                    # Model variance from fit function
+                    data_snr[data_snr>11] = 11
+                    data_snr[ data_snr<1.01] = 1.01
+                    variance_estimated = variance_function(data_snr, *coef)
+                    weights = 1. / variance_estimated
+                    mean = np.average((p1d_table[c][select]), weights=weights)
+                    error = np.sqrt(1. / np.sum(weights))
+                    if output_snrfit is not None and c=='Pk':
+                        snrfit_table[index, 0:4] = [zbin_centers[izbin], (kbin+kbin_edges[ikbin+1])/2.,
+                                                    coef[0], coef[1]]
+                        snrfit_table[index, 4:] = standard_dev
+
+                elif weight_method=='simple_snr':
+                    # for forests with snr>snr_limit (hardcoded to 4 as of now), 
+                    # the weight is fixed to (snr_limit - 1)**2 = 9
+                    snr_limit = 4 
+                    forest_snr = p1d_table['forest_snr'][select]
+                    w, = np.where(forest_snr <= 1)
+                    if len(w)>0: raise RuntimeError('Cannot add weights with SNR<=1.')
+                    weights = (forest_snr - 1)**2
+                    weights[forest_snr>snr_limit] = (snr_limit - 1)**2
+                    mean = np.average((p1d_table[c][select]), weights=weights)
+                    # Need to rescale the weights to find the error:
+                    #   weights_true = weights * (N - 1) / alpha
+                    alpha = np.sum(weights * ((p1d_table[c][select] - mean)**2))
+                    error = np.sqrt(alpha / (np.sum(weights) * (N - 1)))
+
+                elif weight_method=='no_weights':
+                    mean = np.average((p1d_table[c][select]))
+                    error = np.std((p1d_table[c][select])) / np.sqrt(N-1)  # unbiased estimate: N-1
+
+                else:
+                    raise ValueError("Option for 'weight_method' argument not found")
```

Comment:
in principle one should have another set of z and k loop to compute the correlation matrix (only looping until izbin (and for the current z-bin only until ikbin-1) for symmetry reasons), with the same weighting as applyed here...

this should then be something as simple as `Cji=Cij=np.mean( (p1d_table[c][select])-mean) *  (p1d_table[c][select2]-meanP1D_table['mean'+c][index2])))/np.sqrt((N-1)*(meanP1D_table['N'][index]-1))` to get the diagonals match the variance (and `C/error[i]/error[j]` being the correlation matrix)

Reply:
Yes. We agreed yesterday to have another PR for that, to start when this one is done

Reply:
great! Then lets do this.